code/modules/mob/living/carbon/human/life.dm

```diff
@@ -169,11 +172,18 @@
 	if(species.vision_organ)
 		vision = internal_organs_by_name[species.vision_organ]
 
-	if(!vision) // Presumably if a species has no vision organs, they see via some other means.
+	if(species.vision_organ && !vision) // eyes have been removed via surgery or other means
+		vision = "removed" // as nothing is done with it later in this proc this is an okay fix, should probably be fixed better later

```

Comment:
No. Absoloutely not okay.

vision is of type obj/item/organ, purposefully giving it a value that is not of this type is literally asking for runtime errors. You can literally not use this crummy hack, use no new variables, and just use if-statements like this:

```
	if (!vision)
		if (species.vision_organ)
			eye_blind = 1
			blinded = 1
			eye_blurry = 1
		else
			eye_blind = 0
			blinded = 0
			eye_blurry = 0
	else if (vision.is_broken())
		eye_blind = 1
		blinded = 1
		eye_blurry = 1
```